src/App.tsx

```diff
@@ -15,36 +18,122 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const getPreparedGoods = (goods: string[], sortOptions: TypeSort) => {
+  const preparedGoods = [...goods];
+
+  if (sortOptions.filterType) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortOptions.filterType) {
+        case FilterTypes.byAlphabet:
+          return good1.localeCompare(good2);
+        case FilterTypes.byLength:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (sortOptions.isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+};
+
 export const App: React.FC = () => {
+  const [filterInstruction, setFilterInstruction] = React.useState({
+    filterType: FilterTypes.default,
+    isReversed: false,
+  });
+
+  const isAnyFilterApplied =
+    filterInstruction.filterType || filterInstruction.isReversed;
+
+  const visibleGoods = getPreparedGoods(goodsFromServer, filterInstruction);
+
+  const handleSetFilterType = (filterTypeName: FilterTypes) => {
```

Comment:
The function's name should start with a verb to clearly indicate what it does. Consider renaming `handleSetFilterType` to `setFilterType` or `updateFilterType` to follow the [CODE STYLE] checklist.